Chore/CI Actions

clr-cera opened the pull request:
# CI Actions for linting and formatting using ruff and black

This pull request inserts github actions for linting and formatting checks using ruff and black.

Fixes #29

## How to ensure format and lint
You need the ruff and black python packages installed. The uv package manager can be used to install and run them automatically as following:

To format the current directory:

```bash
uvx black .
```

To check the format:

```bash
uvx black --check .
```

To check lint:

```bash
uvx ruff check
```

clr-cera committed:
repo: add black and ruff github actions

Added ruff and black github actions, they are separate as one is
independent from the other (both can show issues, so both need to be ran
even if the other has failed), and the yml file describing each one is
very simple and standard.

clr-cera committed:
format: format the current codebase to standard black formmater

Used black to format the file kobra.py.

clr-cera committed:
fix: black action call to stable version

clr-cera committed:
code: join CI jobs in single workflow

clr-cera commented:
I need to fix the ci workflow running twice in pull requests, once for push trigger and once for pull_request trigger

clr-cera committed:
fix: only run ci on pr

Only run ci on PR, as main branch should be protected, and it is only a
validation github action

clr-cera commented:
Now it should be right!! 🥳

otaviobia commented:
LGTM!
I suggest a follow-up Issue/PR to update [manual.md](https://github.com/fossguild/naja/blob/main/docs/manual.md) with the new requirements introduced in #7 and #30.

clr-cera committed:
Merge branch 'main' into chore/CI-actions

clr-cera committed:
format after fixing conflicts

bdogabriel commented:
I think it would be nice if the CI followed as close as possible the local pre-commit setup. Do you think you can change it here? I could also change some stuff in #34. What do you think would be the ideal setup?

clr-cera commented:
I agree!! I chose ruff as it is the most performant linter I know. 
The pre-commit hooks also use black right? So that is already compatible I believe (would only need to check if uses same version)
And as I know, ruff linter already checks for import sorting, so I believe that if ruff is chosen isort is not needed, or is there another use for isort??

bdogabriel commented:
Perfect! I updated the pre-commit PR to use ruff, so isort is not necessary anymore. Thanks for the help :)

Edit: I can't approve this PR, I think it's because it was opened before I had write access, idk

clr-cera commented:
Ohhh perfect!!

And damn, we will have to wait for another review then, and as soon as I have some time I will review the pre commit pr too!

clr-cera committed:
docs: documenting development requirements

uv is optional as all packages can be installed system wise but ruff and
black are needed for pull request merging.

CONTRIBUTING and conventions were also updated to describe this need

clr-cera commented:
> LGTM! I suggest a follow-up Issue/PR to update manual.md with the new requirements introduced in #7 and #30.

@otaviobia I have just made a commit documenting it! (Have also updated CONTRIBUTING.md and conventions.md)

clr-cera committed:
Merge branch 'main' into chore/CI-actions

LiviaLelis reviewed (approved):
Seems good :D
